server/models.py

```diff
@@ -0,0 +1,110 @@
+import secrets
+from config import db, bcrypt, MetaData
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(100), nullable=False)
+    email = db.Column(db.String(100), unique=True, nullable=False)
+    _password = db.Column(db.String)
+    financial_goals = db.relationship('FinancialGoal', backref='user', lazy=True)
+    accounts = db.relationship('Account', backref='user', lazy=True)
+    family_id = db.Column(db.Integer, db.ForeignKey('family.id'))
+    insights = db.relationship('Insight', backref='user', lazy=True)
+
+    def validate_email(self, key, address):
+        pattern = "([\w\.]+@[A-Za-z]+\.[A-Za-z]+)"
+        if not re.match(pattern, address):
```

Comment:
I took some logic from the user model here https://github.com/JWehder/Learn-Link/blob/main/server/models.py#L55